backend/src/services/v2/file.ts

```diff
@@ -0,0 +1,82 @@
+import { putObjectStream } from '../../clients/s3.js'
+import authorisation, { ModelAction } from '../../connectors/v2/authorisation/index.js'
+import FileModel from '../../models/v2/File.js'
+import { UserDoc } from '../../models/v2/User.js'
+import config from '../../utils/v2/config.js'
+import { Forbidden, NotFound } from '../../utils/v2/error.js'
+import { longId } from '../../utils/v2/id.js'
+import { getModelById } from './model.js'
+
+export async function uploadFile(user: UserDoc, modelId: string, name: string, mime: string, stream: ReadableStream) {
```

Comment:
what is 'mime'?

Reply:
Mime types refer to what _type_ of file we're going to expect. There's a list of common ones [here](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types). Whilst there are libraries to type _common_ mime types, you can always make your own so it's best to leave as string.v